hera_sim/simulate.py

```diff
@@ -434,4 +435,75 @@ def add_xtalk(self, model='gen_whitenoise_xtalk', bls=None, **kwargs):
                 vis=self.data.data_array[blt_ind, 0, :, pol_ind],
                 xtalk=xtalk
             )
+    
+    # make a dictionary whose values point to the various methods
+    # used to add different simulation components
+    SIMULATION_COMPONENTS = { 'noiselike_eor':add_eor,
```

Comment:
I had to move this dictionary to some point *after* the functions being pointed to were defined. If we try to make a pointer to an object which does not exist yet, then python is going to throw an error. In fact, the previous build failed precisely because of this. If you're not sold, then try out this simple code:
```
class TestClass():
    FUNCTIONS = {'func1':func1, 'func2':func2}
    def func1(self):
        return None
    def func2(self):
        return None
```
If you try to run this, you'll get a `NameError`; however, if you move the `FUNCTIONS = ...` line to a point after `func1` and `func2` have been defined, then everything's fine.

Reply:
yup, that makes perfect sense. If you really want it at the top (and that _would_ be nice), you can replace the functions/methods with _strings_ giving their names, and then just do `getattr(self, stringname)` when you access it.

Reply:
I like that solution. I implemented it in the latest commit.